tiledb/sm/filesystem/win.cc

```diff
@@ -586,10 +581,15 @@ Status Win::write_at(
   // file handle. Instead, we use the OVERLAPPED struct to specify an offset at
   // which to write. Note that the file handle does not have to be opened in
   // "overlapped" mode (i.e. async writes) to do this.
-  unsigned long bytes_written = 0;
   uint64_t byte_idx = 0;
   const char* byte_buffer = reinterpret_cast<const char*>(buffer);
-  while (buffer_size > constants::max_write_bytes) {
+  uint64_t remaining_bytes_to_write = buffer_size;
+  while (remaining_bytes_to_write > 0) {
+    DWORD bytes_to_write =
+        remaining_bytes_to_write > std::numeric_limits<DWORD>::max() ?
```

Comment:
Making this limit Windows-specific seems reasonable, but if we are not going to use the existing constant here, we should document the variance next to the constant definition, or rename it `..._posix` to avoid future confusion.

Reply:
Just a heads up, the other VFS refactor PR removes the use of `constants:max_write_bytes` as well.